gpflow/base.py

```diff
@@ -39,15 +39,17 @@ def __init__(self,
                  *,
                  transform: Optional[Transform] = None,
                  prior: Optional[Prior] = None,
+                 prior_on_x: bool = True,
                  trainable: bool = True,
                  dtype: Optional[DType] = None,
                  name: Optional[str] = None):
         """
-        Unconstrained parameter representation.
-        According to standard terminology `y` is always transformed representation or,
-        in other words, it is constrained version of the parameter. Normally, it is hard
-        to operate with unconstrained parameters. For e.g. `variance` cannot be negative,
-        therefore we need positive constraint and it is natural to use constrained values.
+        Representation of a constrained parameter.
+        According to standard terminology `x` denotes the transformed representation or,
```

Comment:
I don't think it's sufficiently standard to not have to link to anything - e.g. we might say that we are using tfp.bijectors and therefore follow their terminology - according to https://www.tensorflow.org/probability/api_docs/python/tfp/bijectors/Bijector `x` is the input, and `y` is the transformed representation (`y = bijector.forward(x)`).

Reply:
Agreed, I've reduced the usage of x, and reversed the definition (x, y) where it does still appear, in def log_prior()